plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployConsole.java

```diff
@@ -0,0 +1,33 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
+
+import org.eclipse.ui.console.MessageConsole;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.standard.StandardDeployJob;
+import com.google.cloud.tools.eclipse.ui.util.MessageConsoleUtilities.ConsoleFactory;
+
+public class DeployConsole extends MessageConsole {
+
+  private static final String TYPE = "com.google.cloud.tools.eclipse.appengine.deploy.consoleType";
+  private StandardDeployJob job;
+
+  public DeployConsole(String name) {
+    super(name, null);
+    setType(TYPE);
+  }
+
+  public StandardDeployJob getJob() {
+    return job;
+  }
+
+  public void setJob(StandardDeployJob deployJob) {
+    this.job = deployJob;
```

Comment:
In the current implementation it is not possible, as the console is created on the UI thread as part of the command handler, and the console page creation also happens on the UI thread via `asyncExec`.
Nevertheless, it's definitely a fragile construction, so I'll look into it.